randovania/interface_common/options.py

```diff
@@ -139,10 +140,25 @@ def _save_to_disk(self):
         data_to_persist = self._serialize_fields()
 
         self._data_dir.mkdir(parents=True, exist_ok=True)
-        with self._data_dir.joinpath("config.json").open("w") as options_file:
+
+        # Write to a separate file, so we don't corrupt the existing one in case we unexpectedly
+        # are unable to finish writing the file
+        new_config_path = self._data_dir.joinpath("config_new.json")
+        with new_config_path.open("w") as options_file:
             json.dump(data_to_persist, options_file,
                       indent=4, separators=(',', ': '))
 
+        # Unix has atomic rename that overwrites
+        # Windows doesn't, so we need to erase the current config before renaming
+        # This opens a small chance of data loss. We could instead rename to `config_old`,
+        # and when loading we would also check that path. But that's way overkill for this
+        config_path = self._data_dir.joinpath("config.json")
+        if platform.system() == "Windows" and config_path.exists():
+            config_path.unlink()
+
+        # Place the new, complete, config to the desired path
+        new_config_path.rename(config_path)
+
```

Comment:
I think we could just use `Path.replace`.
https://docs.python.org/3/library/pathlib.html#pathlib.Path.replace